unix/vncserver/vncsession.c

```diff
@@ -511,14 +519,28 @@ main(int argc, char **argv)
 
     const char *username, *display;
 
-    if ((argc != 3) || (argv[2][0] != ':')) {
-        fprintf(stderr, "Syntax:\n");
-        fprintf(stderr, "    %s <username> <display>\n", argv[0]);
-        return EX_USAGE;
+    int opt, forking = 1;
+    if (argc == 4) {
```

Comment:
The count should be checked after `getopt()` has done its thing. Let `getopt()` figure out all the flags, and then we look at the remaining arguments to see if they make sense.

Reply:
Hopefully the updated PR does what you ask. It is an improvement because now there is no duplication of the check for the ':' character at the start of the last argument which is for the display.